src/clr-angular/forms/datepicker/date-input.ts

```diff
@@ -61,7 +61,28 @@ export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implement
             this.populateContainerServices();
         }
         this.initializeSubscriptions();
+        this.processInitialInputs();
+    }
+
+    /**
+     * Process the inputs initialized by the user which were missed
+     * because of late subscriptions or lifecycle method calls.
+     */
+    private processInitialInputs(): void {
         this.processUserDateObject(this.dateValueOnInitialLoad);
+
+        // Handle Inital Value from Reactive Forms
+        // TODO: We are repeating this logic at multiple places. This makes me think
+        // if this class should have implemented the ControlValueAccessor interface.
+        // Will explore that later and see if its a cleaner solution.
+        if (this._ngControl && this._ngControl.value) {
+            const date: Date = this._dateIOService.isValidInput(this._ngControl.value);
+            if (date) {
+                this._dateNavigationService.selectedDay =
+                    new DayModel(date.getFullYear(), date.getMonth(), date.getDate());
+                this.writeDateStrToInputField(this._ngControl.value);
```

Comment:
That's a lot of code duplication with `processUserDateObject()`. It's actually the same method, but `processUserDateObject()` starts by converting the Date object to a string, then does exactly the same as this.

Please rewrite these two methods so that they share the common code.

Reply:
@youdz: done.